vars/signArtifacts.groovy

```diff
@@ -30,8 +30,9 @@ void call(Map args = [:]) {
 }
 
 void importPGPKey(){
-    if( !fileExists("$WORKSPACE/opensearch.pgp")) {
-        sh("curl -SL https://artifacts.opensearch.org/publickeys/opensearch.pgp -o $WORKSPACE/opensearch.pgp")
-        sh("gpg --import $WORKSPACE/opensearch.pgp")
+    keyPath = "$WORKSPACE/opensearch.pgp"
+    if( !fileExists("${keyPath}")) {
+        sh("curl -SL https://artifacts.opensearch.org/publickeys/opensearch.pgp -o ${keyPath}")
```

Comment:
Since you import every time, how about never storing this key?
https://www.reddit.com/r/GnuPG/comments/4oz7zd/possible_to_import_gpg_key_from_a_url/

```
curl -sSL http://my.key.url | gpg --import -
```